### PR TITLE
MOEN-30132 Non Intrusive Nudges Support 

### PR DESCRIPTION
**Pull request by kkgowtham-engg-sdk**
### Jira Ticket
https://moengagetrial.atlassian.net/browse/MOEN-30132

### Description
Non Intrusive Nudges Support

### CODE REVIEW

File: core/moengage_flutter_platform_interface/lib/src/moengage_flutter_platform_interface.dart [hunk 1]
```diff
@@ -209,4 +210,11 @@ abstract class MoEngageFlutterPlatform extends PlatformInterface {
   /// @since 1.1.0
   Future<UserDeletionData> deleteUser(String appId) =>
       throw UnimplementedError('deleteUser() not implemented for Platform');
+
+  /// Try to show a non-intrusive In-App nudge.
+  /// [position] - Nudge InApp Position of type [MoEngageNudgePosition]
+  /// [appId] - MoEngage App ID
+  /// @since TODO: Update Version
+  void showNudge(MoEngageNudgePosition position, String appId) =>
+      throw UnimplementedError('showNudge() not implemented for Platform');
```
Comment (RakshithaAcharya) [inline core/moengage_flutter_platform_interface/lib/src/moengage_flutter_platform_interface.dart:219]: @msoumya-engg-sdk what is the purpose of this method

**Review by RakshithaAcharya — CHANGES_REQUESTED** [review]

File: core/moengage_flutter_platform_interface/lib/src/moengage_flutter_platform_interface.dart (hunk 1 above)
Comment (msoumya-engg-sdk) [reply core/moengage_flutter_platform_interface/lib/src/moengage_flutter_platform_interface.dart:219]: This is the interface method which is overridden by each platform with platform specific implementation.

**Review by RakshithaAcharya — APPROVED** [review]

File: cards/moengage_cards_android/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # MoEngage Cards Android Plugin
 
+# Next Release
+
+## Next Version
+- Support for `moe-android-sdk` version `12.10.04` and above
```
Comment (arshiya-moengage) [inline cards/moengage_cards_android/CHANGELOG.md:6]: The version has to be updated

File: cards/moengage_cards_android/android/build.gradle [hunk 3]
```diff
@@ -50,7 +50,7 @@ android {
 dependencies {
 
     api("com.moengage:plugin-base-cards:1.1.0")
-    compileOnly("com.moengage:moe-android-sdk:12.10.01")
+    compileOnly("com.moengage:moe-android-sdk:12.10.04")
```
Comment (arshiya-moengage) [inline cards/moengage_cards_android/android/build.gradle:53]: version is outdated

File: core/moengage_flutter_android/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # MoEngage Flutter Android Plugin
 
+# Next Release
+
+## Next Version
+- Non-intrusive Nudges support
+- Support for `moe-android-sdk` version `12.10.04` and above
```
Comment (arshiya-moengage) [inline core/moengage_flutter_android/CHANGELOG.md:7]: update the version

File: core/moengage_flutter_android/android/build.gradle [hunk 5]
```diff
@@ -50,8 +50,8 @@ android {
     }
 }
 dependencies {
-    compileOnly("com.moengage:moe-android-sdk:12.10.01")
-    compileOnly("com.moengage:inapp:7.1.0")
+    compileOnly("com.moengage:moe-android-sdk:12.10.04")
```
Comment (arshiya-moengage) [inline core/moengage_flutter_android/android/build.gradle:53]: update the versions

File: core/moengage_flutter_web/lib/moengage_flutter_web.dart
Comment (arshiya-moengage) [inline core/moengage_flutter_web/lib/moengage_flutter_web.dart:1]: Revert these changes

File: example/android/settings.gradle [hunk 6]
```diff
@@ -22,7 +22,7 @@ dependencyResolutionManagement {
     versionCatalogs {
         libs {
             create("moengage") {
-                from("com.moengage:android-dependency-catalog:3.2.1")
+                from("com.moengage:android-dependency-catalog:3.2.5")
```
Comment (arshiya-moengage) [inline example/android/settings.gradle:25]: Update the version

File: geofence/moengage_geofence_android/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,10 @@
 # MoEngage Geofence Android Plugin
 
+# Next Release
+
+## Next Version
+- Support for `moe-android-sdk` version `12.10.04` and above
```
Comment (arshiya-moengage) [inline geofence/moengage_geofence_android/CHANGELOG.md:6]: update the native version

File: geofence/moengage_geofence_android/android/build.gradle [hunk 8]
```diff
@@ -49,7 +49,7 @@ android {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-    compileOnly("com.moengage:moe-android-sdk:12.10.01")
+    compileOnly("com.moengage:moe-android-sdk:12.10.04")
```
Comment (arshiya-moengage) [inline geofence/moengage_geofence_android/android/build.gradle:52]: update the version

File: inbox/moengage_inbox_android/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,10 @@
 # MoEngage Inbox Android Plugin
 
+# Next Release
+
+## Next Version
+- Support for `moe-android-sdk` version `12.10.04` and above
```
Comment (arshiya-moengage) [inline inbox/moengage_inbox_android/CHANGELOG.md:6]: update version

File: inbox/moengage_inbox_android/android/build.gradle [hunk 10]
```diff
@@ -49,7 +49,7 @@ android {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-    compileOnly("com.moengage:moe-android-sdk:12.10.01")
+    compileOnly("com.moengage:moe-android-sdk:12.10.04")
```
Comment (arshiya-moengage) [inline inbox/moengage_inbox_android/android/build.gradle:52]: update version

File: core/moengage_flutter_web/lib/moengage_flutter_web.dart
Comment (kkgowtham-engg-sdk) [reply core/moengage_flutter_web/lib/moengage_flutter_web.dart:1]: Reverted.

File: core/moengage_flutter_android/android/build.gradle [hunk 11]
```diff
@@ -48,11 +48,15 @@ android {
...
+        buildConfig true
+    }
 }
 dependencies {
-    compileOnly("com.moengage:moe-android-sdk:12.10.01")
-    compileOnly("com.moengage:inapp:7.1.0")
-    api("com.moengage:plugin-base:3.4.0")
+    compileOnly("com.moengage:moe-android-sdk:13.00.00")
+    compileOnly("com.moengage:inapp:8.0.0")
+    api("com.moengage:plugin-base:4.0.0")
```
Comment (kkgowtham-engg-sdk) [inline core/moengage_flutter_android/android/build.gradle:59]: TODO: Confirm Plugin Base Versions before Merging

File: inbox/moengage_inbox_android/android/build.gradle (hunk 10 above)
Comment (kkgowtham-engg-sdk) [reply inbox/moengage_inbox_android/android/build.gradle]: Updated

File: geofence/moengage_geofence_android/android/build.gradle (hunk 8 above)
Comment (kkgowtham-engg-sdk) [reply geofence/moengage_geofence_android/android/build.gradle]: Updated

File: geofence/moengage_geofence_android/CHANGELOG.md (hunk 7 above)
Comment (kkgowtham-engg-sdk) [reply geofence/moengage_geofence_android/CHANGELOG.md]: Updated

File: example/android/settings.gradle (hunk 6 above)
Comment (kkgowtham-engg-sdk) [reply example/android/settings.gradle]: updated

File: core/moengage_flutter_android/android/build.gradle (hunk 5 above)
Comment (kkgowtham-engg-sdk) [reply core/moengage_flutter_android/android/build.gradle]: Updated

File: core/moengage_flutter_android/CHANGELOG.md (hunk 4 above)
Comment (kkgowtham-engg-sdk) [reply core/moengage_flutter_android/CHANGELOG.md]: Updated

File: cards/moengage_cards_android/android/build.gradle (hunk 3 above)
Comment (kkgowtham-engg-sdk) [reply cards/moengage_cards_android/android/build.gradle]: Updated

File: cards/moengage_cards_android/CHANGELOG.md (hunk 2 above)
Comment (kkgowtham-engg-sdk) [reply cards/moengage_cards_android/CHANGELOG.md]: Updated

**Pull request merged**